slug.go

```
@@ -382,119 +392,92 @@ func (p *Packer) Unpack(r io.Reader, dst string) error {
-		fh, err := os.Create(path)
+		fh, err := os.Create(info.Path)
 		if err != nil {
 			// This mimics tar's behavior wrt the tar file containing duplicate files
 			// and it allowing later ones to clobber earlier ones even if the file
-			// has perms that don't allow overwriting.
+			// has perms that don't allow overwriting. The file permissions will be restored
+			// once the file contents are copied.
 			if os.IsPermission(err) {
-				os.Chmod(path, 0600)
-				fh, err = os.Create(path)
+				os.Chmod(info.Path, 0600)
```

golangci-lint (1.64.8)

Error return value of `os.Chmod` is not checked

(errcheck)